File: Java/Kata/Kata.java
public class Kata{

public static int numberOfFactors(int number){

int counter = 0;
int numbertocheck = 0;
	for (int count = 1; count <= number; count++){
	if (number % count == 0){
		counter++;
	}
	
}
	return counter;

}



public static Boolean isprime(int number){
	 if(number <= 1){
	   return false;
}
	for(int cont = 2; cont*cont <= number; cont++);
	if (number % 2 == 0){
	return false;
}
	return true;
}
	
	public static boolean isOdd(int number) {
	if(number % 2 != 0) {
		return true;
	}
	if(number % 2 == 0) {
		return false;
	}
	return false;
}




	public static boolean isEven(int number) {
	if(number % 2 == 0) {
		return true;
	}
	return false;
}

	

	public static int subtract(int firstNumber, int secondNumber) {
	int difference = firstNumber - secondNumber;
	if(difference < 0) {
		difference *= -1;
	}
	return difference;
}

	public static boolean isSquare(int number) {
	int divisor = 1;
	int Count = 0;
	while(divisor <= number) {
		int divide = number % divisor;
		if(divide == 0){
			Count += 1;
		}
		divisor += 1;
	}

	if(Count % 2 != 0) {
		return true;
	}

	return false;
}
 
	public static long factorialOf(int number){
	int factorial = 1;
	int counter = 1;
	while(counter <= number) {
		factorial *= counter;
		counter++;
	}
	return factorial;
}

	public static long squareOf(int number) {
	int sum = number * number;
	return sum;
}

	public static float divide(int firstNumber, int secondNumber) {
	if(secondNumber == 0) {
		return 0;
	} else {
		return firstNumber / secondNumber;
}

}

	public static boolean isPalindrome(int number) {
	int firstDigit = number / 10000;
	int secondDigit = (number % 10000) / 1000;
	int thirdDigit = (number % 1000) / 100;
	int fourthDigit = (number % 100) / 10;
	int fifthDigit = number % 10;
	if(firstDigit == fifthDigit & secondDigit == fourthDigit) {
		return true;
}	
	return false;
}
}

File: Python/Kata/Kata.py
def factor_of(number):
	divisor = 1
	count_Factors = 0
	while divisor <= number :
		sum = number % divisor
		if sum == 0:
			countFactors += 1
		divisor += 1
	return count_Factors
print(factor_of(30))

def is_prime_number(number):
	divisor = 1
	Count = 0
	while divisor <= number :
		divide = number % divisor
		if divide == 0:
			Count += 1
		divisor += 1
	if Count == 2:
		return True
	if Count > 2:
		return False
print(is_prime_number(30))


def is_even(number):
	if number % 2 == 0:
		return True
	if number % 2 != 0:
		return False
print(is_even(30))


def subtract(first_number, second_number):
	sum = second_number - first_number
	if first_number > second_number:
		sum = sum * -1
	return answer
print(subtract(30))


def divide(first_number, second_number):
	if second_number == 0:
		return 0
	if second_number != 0:
		return first_number / second_number
print(divide(30))


def is_square(number):
	divisor = 1
	Count = 0
	while divisor <= number :
		divide = number % divisor
		if divide == 0 :
			Count += 1;
		divisor += 1;
	

	if factorCount % 2 != 0 :
		return True
	return False
print(is_square(30))

def is_palindrome(number):
	if number <= 99999 and number >= 10000:
		digit_1 = number // 10000
		digit_2 = (number % 10000) // 1000
		digit_3 = (number % 1000) // 100
		digit_4 = (number % 100) // 10
		digit_5 = number % 10

	new_number = str(digit_5) + str(digit_4) + str(digit_3) + str(digit_2) + str(digit_1)


	if int(new_number) == number:
		return True
	return False
print(is_palindrome(30))


def factorial_of(number):
	factorial = 1
	counter = 1
	while counter <= number :
		factorial *= counter
		counter += 1
	return factorial
print(factorial_of(30))


def square_of(number):
	return number ** 2
print(square_of(30))
